Library/Homebrew/software_spec.rb

```diff
@@ -307,13 +308,23 @@ def initialize(formula, spec)
     filename = Filename.create(formula, tag, spec.rebuild)
     @resource.url("#{spec.root_url}/#{filename.bintray}",
                   select_download_strategy(spec.root_url_specs))
+    if @resource.download_strategy == CurlDownloadStrategy && !spec.default_root_url.nil?
```

Comment:
```suggestion
    if @resource.download_strategy == CurlDownloadStrategy && spec.default_root_url.present?
```

Reply:
Can you explain the `spec.default_root_url` bit a bit, I'm not sure I understand when that is/isn't present?